stan/math/rev/core/vari.hpp

```diff
@@ -236,7 +350,7 @@ class vari_value<T, require_all_t<is_plain_type<T>, is_eigen_dense_base<T>>>
   using PlainObject = plain_type_t<T>;
   using value_type = PlainObject;  // The underlying type for this class
   using eigen_scalar = value_type_t<PlainObject>;  // A floating point type
-  using eigen_map = Eigen::Map<PlainObject, Eigen::Aligned8,
+  using eigen_map = Eigen::Map<PlainObject, Eigen::Unaligned,
```

Comment:
Does the arena allocator not guarantee some alignment? Hmm a comment says so, but I can't find any code that would ensure that. Anyway if you are using default alignment and stride you don't need to specify them.

Reply:
Yeah the allocator guarantees 8 byte alignment I can fix this up

It looks like Map uses `Unaligned` by default. One thing I was confused by here, does telling Map the alignment tell it "This is _at least_ 8 byte aligned" or that it is exactly 8 byte aligned?

https://eigen.tuxfamily.org/dox/classEigen_1_1Map.html

Reply:
Why does it matter? If some object is aligned to more than 8byte it is at the same time 8byte aligned.

Also if that is so, probably the arena matrix also should tell its maps the data is aligned.

Reply:
> Why does it matter? If some object is aligned to more than 8byte it is at the same time 8byte aligned.

I didn't think it did, but I've never looked under the hood at Eigen here to check if they do something in particular on particular alignments that needed a certain alignment length.

> Also if that is so, probably the arena matrix also should tell its maps the data is aligned.

Yeah in the multiply PR I changed that since we have the same guarantee

Reply:
I ended up having to move these and arena_matrix back to the default `Unaligned` because of an error upstream in the stan tests where I'm getting 

```cpp
rvalue_test: lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/MapBase.h:194: void Eigen::MapBase<Derived, 0>::checkSanity(typename Eigen::internal::enable_if<(Eigen::internal::traits<OtherDerived>::Alignment > 0), void*>::type) const [with T = Eigen::Map<Eigen::Matrix<double, -1, 1>, 8, Eigen::Stride<0, 0> >; Derived = Eigen::Map<Eigen::Matrix<double, -1, 1>, 8, Eigen::Stride<0, 0> >; typename Eigen::internal::enable_if<(Eigen::internal::traits<OtherDerived>::Alignment > 0), void*>::type = void*]: Assertion `( ((internal::UIntPtr(m_data) % internal::traits<Derived>::Alignment) == 0) || (cols() * rows() * minInnerStride * sizeof(Scalar)) < internal::traits<Derived>::Alignment ) && "data is not aligned"' failed.
Aborted (core dumped)
```

Reply:
Lol, the doc on the allocator needs some fixing. Please revert the changes to arena_matrix_as well.